requirements_test.txt

```diff
@@ -1,17 +1,14 @@
-# linters such as flake8 and pylint should be pinned, as new releases
+# linters such as pylint should be pinned, as new releases
 # make new things fail. Manually update these pins when pulling in a
 # new version
 
 # When updating this file, update .pre-commit-config*.yaml too
 asynctest==0.13.0
 black==19.10b0
 codecov==2.0.15
-flake8-docstrings==1.5.0
-flake8==3.7.8
```

Comment:
It's nice to be able to install the correct linters requirements without having to go through pre-commit, ie just by doing `pip install -r ...`.

I'd like to keep the requirements specified here.

Reply:
I wouldn't mind it that much, but we're already failing on the "correct" part. These are now specified in two places, and they're already off sync.

Reply:
Maybe we can sync them with `script/gen_requirements_all.py`?

Reply:
That should be doable, parse from .pre-commit-config-all.yaml and output to the .txt. Will add a dependency on yaml parser to gen_requirements_all.py, not sure if that's considered a problem though.

I can have a crack at this is if keeping them is what the final judgement is, but I'd rather not do that yet because I personally think it's 1) unnecessary code and time not that well spent, and 2) would be just a matter of small tweaks to people's habits to run stuff through pre-commit, and 3) results in unnecessarily installed dependencies (duplicate flake8 and friends) to all dev setups.

BTW I think we should actually move more stuff to pre-commit (for example pylint, probably to the pre-commit "all" config so it's not always run for everyone) to avoid similar config duplication. Meaning there will be more duplicate installs.

Reply:
See #28538, it's a useful intermediate step in case we don't want to take all the way here.